crates/storage/src/schema/revision_0027.rs

```diff
@@ -0,0 +1,19 @@
+/// This migration adds a new table storing compiled CASM classes.
+pub(crate) fn migrate(tx: &rusqlite::Transaction<'_>) -> anyhow::Result<()> {
+    use anyhow::Context;
+
+    tx.execute_batch(
+        r"-- Stores compiled CASM for Sierra classes.
+        CREATE TABLE contract_code_casm (
+            hash                BLOB    PRIMARY KEY NOT NULL,
+            compiled_class_hash BLOB    NOT NULL,
+            definition          BLOB    NOT NULL,
```

Comment:
Should we associate the specific compiler version as well? I'm thinking that maybe at some point there is an issue with a specific compiler and we would need to recompile all classes it touched?

Reply:
I'm not 100% sure we will need that. We _only_ use these CASM classes for execution -- so the only failure mode we would care about is when the _output_ of the execution might be different?

And even then I think we should be just able to re-compile _all_ CASM classes. It's quite fast, and there are ~20k (Cairo 0.x) classes on testnet right now so re-compiling Sierra classes shouldn't take too long?

Reply:
Yes but that is currently. That won't really scale once starknet takes off? 

I imagine over time we would have many different compiler's outputs in our database. If we think that simply recompiling them all will be fine, versus only recompiling the problematic ones, then I suppose this is fine as is.

Reply:
I'm not opposed to adding the compiler version. We could add the version of the crate as a string maybe? That should be doable, but involves having a `build.rs` that uses Cargo metadata to define that string.

Reply:
Yeah I'm just thinking of things like transactions. Where we have to fully open the json to decide what variant the tx is. If we had stored the variant as a separate int, it would have made several migrations much faster.